### PR TITLE
Add CSV output

### LINKED ISSUE #255: Add support for raw measurement data
Currently the options for getting raw measurement data out (see below) are not implemented. It would be great to have these for more fine-grained analysis of the runs.

https://github.com/pingcap/go-ycsb/blob/d0a53f45c606f87675cf26748eb611b0630990e9/workloads/workload_template#L121-L137

### PR DESCRIPTION
**Pull request by jeffa5**
- Record start time in measurements
- Adds the CSV measurement type
- Adds configurable writing to file for measurements
- Rework measurement reporting to be handled inside the measurement type (histograms output different structure to csv)
    - This removes the `Info` and `Summary` methods providing a simpler interface
- Add output configuration section to README

Fixes #255

### CODE REVIEW

File: pkg/measurement/measurement.go [hunk 1]
```diff
@@ -32,90 +32,74 @@ type measurement struct {
...
-		lines = append(lines, line)
+	outFile := m.p.GetString(prop.MeasurementRawOutputFile, "")
+	var w *bufio.Writer
+	if outFile == "" {
+		w = bufio.NewWriter(os.Stdout)
+	} else {
+		f, err := os.Create(outFile)
+		if err != nil {
+			panic("failed to create output file: " + err.Error())
+		}
```
Comment (zyguan) [inline pkg/measurement/measurement.go:56]: We'd better add `defer f.Close()` here, although it's only called once via ycsb command.

**Review by zyguan — COMMENTED** [review]
Rest LGTM.

File: pkg/measurement/measurement.go (hunk 1 above)
Comment (jeffa5) [reply pkg/measurement/measurement.go:56]: Added, thanks!

**Review by zyguan — APPROVED** [review]
LGTM

**Pull request merged**